src/components/Layout/Layout.tsx

```diff
@@ -131,13 +134,25 @@ export const Layout = ({
 
   const handleScroll = debounce((e) => {
     const bodyScroll = e.target.documentElement.scrollTop;
-    if (bodyScroll > 50) {
+    if (bodyScroll > 20) {
       document.body.classList.add('scrolled');
     } else if (document.body.classList.contains('scrolled')) {
       document.body.classList.remove('scrolled');
     }
   });
 
+  const handleMenuToggle = () => {
```

Comment:
Added this function to handle the open/close of the menu so that we can also set focus for keyboard navigation.

The focus on `sidebarMenuButtonRef` is set to a 0 second timeout because before toggling the menu to open, this element is within a `display: none` and the browser won't allow focusing of it (even with settings the `.focus()` to occur after `toggleMenuOpen(true)`).

I'm not sure if there is a better way to handle that? I could not use `display: none` and hide the closed menu through other means but then that complicates handling the tab-ability of the hidden menu links.

Reply:
Not sure if this is any better but it is probably more reacty but I think you could use a useEffect, something like
```
useEffect(() => {
    if (menuOpen) {
      sidebarMenuButtonRef?.current?.focus();
    }
  }, [menuOpen])
  ```
  not a blocker for me though if you prefer the timeout (or if that is just better)

Reply:
I tried to get that to work (with and without the timeout) and with using useEffect() but for some reason it still won't move focus. I know the browser won't let you focus things that are wrapped in a display: none, so maybe theres some order of events that aren't happening at the right time? Like the DOM isn't reconciled in time for the browser to say it can be focused? Not sure :( I left it as is for now.